src/checkers/chromiumchecker.py

```diff
@@ -0,0 +1,183 @@
+        async with self.session.get(url, params=self._UPDATE_PY_PARAMS) as response:
+            result = await response.text()
+
+        update_py = base64.b64decode(result).decode("utf-8")
+        revision = self._CLANG_REVISION_RE.search(update_py).group(1)
+        sub_revision = self._CLANG_SUB_REVISION_RE.search(update_py).group(1)
+
+        return LLVMVersionUtilsMixin.Version(revision, sub_revision)
+
+
+class LLVMGitComponent(Component, LLVMVersionUtilsMixin):
+    NAME = "llvm-git"
+    DATA_CLASS = ExternalGitRepo
+
+    _LLVM_REPO_URL = "https://github.com/llvm/llvm-project"
+
+    async def check(self) -> None:
+        assert isinstance(self.external_data, ExternalGitRepo)
+
+        llvm_version = await self.get_llvm_version()
+
+        new_version = await ExternalGitRef(
+            url=self.external_data.current_version.url,
+            commit=llvm_version.revision,
+            tag=None,
+            branch=None,
+            version=self.latest_version,
+            timestamp=None,
+        ).fetch_remote()
+        self.external_data.set_new_version(new_version)
+
+
+class LLVMPrebuiltComponent(Component, LLVMVersionUtilsMixin):
+    NAME = "llvm-prebuilt"
+    DATA_CLASS = ExternalDataSource
+
+    _PREBUILT_URL_FORMAT = (
+        "https://commondatastorage.googleapis.com"
+        "/chromium-browser-clang/Linux_x64/clang-{revision}-{sub_revision}.tgz"
+    )
+
+    async def check(self) -> None:
+        assert isinstance(self.external_data, ExternalDataSource)
+
+        llvm_version = await self.get_llvm_version()
+
+        latest_url = self._PREBUILT_URL_FORMAT.format(
+            revision=llvm_version.revision, sub_revision=llvm_version.sub_revision
+        )
+        await self.update_external_source_version(latest_url)
+
+
+class ChromiumChecker(Checker):
+    CHECKER_DATA_TYPE = "chromium"
+    SUPPORTED_DATA_CLASSES = [ExternalData, ExternalGitRepo]
+
+    _COMPONENTS = {c.NAME: c for c in Component.__subclasses__()}
+
+    _CHROMIUM_VERSIONS_URL = "https://omahaproxy.appspot.com/all.json"
+    _CHROMIUM_VERSIONS_PARAMS = {"os": "linux", "channel": "stable"}
+
+    async def _get_latest_chromium(self) -> str:
+        async with self.session.get(
+            self._CHROMIUM_VERSIONS_URL, params=self._CHROMIUM_VERSIONS_PARAMS
+        ) as response:
+            result = await response.json()
+
+        assert len(result) == 1, result
+        assert len(result[0]["versions"]) == 1, result
+        return result[0]["versions"][0]["current_version"]
+
+    async def check(self, external_data: t.Union[ExternalData, ExternalGitRepo]):
+        assert self.should_check(external_data)
+
+        component_name = external_data.checker_data.get(
+            "component", ChromiumComponent.NAME
+        )
+        if component_name not in self._COMPONENTS:
+            log.error(f"Invalid component: {component_name}")
+            external_data.state = ExternalBase.State.BROKEN
+            return
```

Comment:
Most other checkers consider malformed metadata fatal, so if the user provided invalid `component` - IMO it's better to raise exception. So, I think this check is redundant.